app/controllers/HaveSecondContactController.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.HaveSecondContactFormProvider
+import models.{AffinityType, CheckMode, UserAnswers}
+import navigation.ContactDetailsNavigator
+import pages.{ContactNamePage, HaveSecondContactPage, SecondContactNamePage}
+import play.api.i18n.{I18nSupport, Messages, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.HaveSecondContactView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class HaveSecondContactController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  navigator: ContactDetailsNavigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: HaveSecondContactFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: HaveSecondContactView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form = formProvider()
+
+  def onPageLoad(): Action[AnyContent] = (identify andThen getData() andThen requireData) {
+    implicit request =>
+      val preparedForm = request.userAnswers.get(HaveSecondContactPage) match {
+        case Some(value) => form.fill(value)
+        case None =>
+          request.userAnswers.get(SecondContactNamePage) match {
+            case Some(_) => form.fill(true)
```

Comment:
Is this correct will that work right for empty  ""

Reply:
Yeah, it's the same logic the check details page uses for answering that question because that question should always be answered, but isn't something that's retrieved from the backend it's just assumed from if second contact details exist, so on the first view of this page it isn't in UserAnswers.

Before I added that logic, the check details page would show it as being answered No, but when you clicked the change link it wouldn't be answered on the page.